posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -259,6 +262,61 @@ def _append_debug_column_to_pyarrows_table(table: pa.Table, load_id: int) -> pa.
     return table.append_column("_ph_debug", column)
 
 
+def should_partition_table(delta_table: deltalake.DeltaTable | None, schema: ExternalDataSchema) -> bool:
+    if not schema.is_incremental:
+        return False
+
+    if delta_table is None:
+        return True
+
+    delta_schema = delta_table.schema().to_pyarrow()
+    if PARTITION_KEY in delta_schema.names:
+        return True
+
+    return False
+
+
+def append_partition_key_to_table(
+    table: pa.Table, incremental_field: str, incremental_field_type: IncrementalFieldType, logger: FilteringBoundLogger
+) -> pa.Table:
+    if (
+        incremental_field_type != IncrementalFieldType.Date
+        and incremental_field_type != IncrementalFieldType.DateTime
+        and incremental_field_type != IncrementalFieldType.Timestamp
+    ):
+        logger.debug(f"No partition key added due to incremental_field_type={incremental_field_type}")
+        return table
+
+    partition_array: list[str | None] = []
+
+    for value in table.column(incremental_field):
+        parsed_value = safe_parse_datetime(value)
+        if not parsed_value:
+            partition_array.append(None)
+            continue
+
+        if isinstance(parsed_value, pa.TimestampScalar):
+            parsed_value_as_py = cast(Any, parsed_value.as_py())
```

Comment:
Is this required to calm mypy? Casting to `Any` seems like it wouldn't do anything.

Reply:
Yeah, sadly so - typings say it returns an `int` which isn't always true